storage/constraints.go

```diff
@@ -167,3 +168,28 @@ func parseSize(s string) (uint64, bool, error) {
 	}
 	return size, true, nil
 }
+
+// ToString returns a parsable string representation of the storage constraints.
+func ToString(c Constraints) (string, error) {
+	if c.Pool == "" && c.Size <= 0 && c.Count <= 0 {
+		return "", errors.Errorf("must provide one of pool or size or count")
+	}
+
+	result := ""
+	if c.Pool != "" {
+		result = c.Pool
+	}
+	if c.Count > 0 {
+		if result != "" {
+			result += ","
+		}
```

Comment:
Slightly tidier to avoid these tests by appending to a `parts []string` instead, and then use `strings.Join(parts, ",")` at the end.

Reply:
yeah, agree